ansible/roles/gpu/tasks/nvidia-gpu-Flatcar.yaml

```diff
@@ -1,18 +1,197 @@
+
+# Note, this has be to run privileged as the driver module has to be loaded (briefly).
+- name: run the build
+  command: docker run -d --rm --privileged --pid=host -v /run/nvidia:/run/nvidia:shared -v /tmp/nvidia:/var/log -v /usr/lib64/modules:/usr/lib64/modules -v /opt:/opt --name nvidia-driver nvcr.io/nvidia/driver:460.73.01-5.10.32-flatcar
+
+- name: wait for build being done
+  command: "docker logs --tail 1 nvidia-driver"
+  register: build_log
+  retries: 150
+  delay: 10
+  until: "'Done, now waiting for signal' in build_log.stdout"
+
+# FIXME(tillt): consider a proper list of libs to grab - for that we would need a proper dependency tree!
+- name: tar nvidia-libs /usr/lib/x86_64-linux-gnu
+  command: docker exec -i nvidia-driver "/bin/sh" -c "tar czf /opt/lib/x86_64-linux-gnu.tar.gz /usr/lib/x86_64-linux-gnu"
+
+- name: tar nvidia-bins /usr/bin/nvidia*
+  command: docker exec -i nvidia-driver "/bin/sh" -c "tar czf /opt/bin/nvidia-bins.tar.gz /usr/bin/nvidia*"
+
+- name: send signal to terminate build container
+  command: "docker kill -s SIGHUP nvidia-driver"
+
+- name: extract libs into /opt/lib/x86_64-linux-gnu
+  command: "tar xzf /opt/lib/x86_64-linux-gnu.tar.gz --directory /opt/lib --strip-components 2"
+
+- name: remove x86_64-linux-gnu tarball
+  command: "rm /opt/lib/x86_64-linux-gnu.tar.gz"
+
+- name: extract bins into /opt/bin
+  command: "tar xzf /opt/bin/nvidia-bins.tar.gz --directory /opt/bin --strip-components 2"
+
+- name: remove nvidia-bins tarball
+  command: "rm /opt/bin/nvidia-bins.tar.gz"
+
+# FIXME(tillt): consider not packing them in the first place
+- name: kill dead libs in /opt/lib/x86_64-linux-gnu
+  command: |
+    rm /opt/lib/x86_64-linux-gnu/libutil.so         \
+      /opt/lib/x86_64-linux-gnu/libthread_db.so     \
+      /opt/lib/x86_64-linux-gnu/librt.so            \
+      /opt/lib/x86_64-linux-gnu/libresolv.so        \
+      /opt/lib/x86_64-linux-gnu/libnsl.so           \
+      /opt/lib/x86_64-linux-gnu/libnss_compat.so    \
+      /opt/lib/x86_64-linux-gnu/libnss_dns.so       \
+      /opt/lib/x86_64-linux-gnu/libnss_files.so     \
+      /opt/lib/x86_64-linux-gnu/libnss_hesiod.so    \
+      /opt/lib/x86_64-linux-gnu/libzstd.so.1        \
+      /opt/lib/x86_64-linux-gnu/libzstd.so.1.3.3    \
+      /opt/lib/x86_64-linux-gnu/libnss_nis.so       \
+      /opt/lib/x86_64-linux-gnu/libnss_nisplus.so   \
+      /opt/lib/x86_64-linux-gnu/libmvec.so          \
+      /opt/lib/x86_64-linux-gnu/libdl.so            \
+      /opt/lib/x86_64-linux-gnu/libcrypt.so         \
+      /opt/lib/x86_64-linux-gnu/libcidn.so          \
+      /opt/lib/x86_64-linux-gnu/libBrokenLocale.so  \
+      /opt/lib/x86_64-linux-gnu/libz.so             \
+      /opt/lib/x86_64-linux-gnu/libanl.so
+
+- name: run ldconfig
+  command: ldconfig /opt/lib/x86_64-linux-gnu
+
+- name: load nvidia modules
+  command: modprobe -d /opt/nvidia/460.73.01/ -a nvidia nvidia-uvm nvidia-modeset
```

Comment:
Can we not use the `modprobe` module here?

Reply:
No, ansible modprobe only supports loading kernel modules from standard locations - we cant provide a directory.